### PR TITLE
fix: numerous small fixes to allow bigtable-admin

### PR DESCRIPTION
**Pull request by software-dov**
Includes:
* tweaked logic around defining recursive message types
* more sophisticated logic for generating unit tests using recursive
message types
* flattened map-y fields are handled properly
* fixed a corner case where a method has a third-party request object
and flattened fields

### CODE REVIEW

**Comment by software-dov** [comment]
Open review bump

File: tests/unit/schema/wrappers/test_field.py [hunk 1]
```diff
@@ -250,6 +251,35 @@ def test_mock_value_message():
...
 
 
+def test_mock_value_recursive():
+    # The elaborate setup is an unfortunate requirement.
+    file_pb = descriptor_pb2.FileDescriptorProto(
+        name="turtle.proto",
+        package="animalia.chordata.v2",
+        message_type=(
+            descriptor_pb2.DescriptorProto(
+                # It's turtles all the way down ;)
```
Comment (busunkim96) [inline tests/unit/schema/wrappers/test_field.py:261]: 🐢 
🐢 🐢 
🐢 🐢 🐢

**Review by busunkim96 — APPROVED** [review]

**Review by arithmetic1728 — APPROVED** [review]
